src/Flysystem/AssetAdapter.php

```diff
@@ -183,6 +185,8 @@ protected function renderTemplate($template)
             }
         }
 
+        $this->extend('updateAllowedExtensions', $allowedExtensions);
```

Comment:
I kind of feel like you'd be better to extend `File::getAllowedExtensions()` instead - are you not able to do that because it's static?